org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/XtextEditor.java

```diff
@@ -327,6 +329,12 @@ public void doSave(IProgressMonitor progressMonitor) {
 		callback.afterSave(this);
 	}
 	
+	@Override
+	protected void performSave(boolean overwrite, IProgressMonitor progressMonitor) {
+		super.performSave(true, progressMonitor);
```

Comment:
This is not something we can merge. Always enforcing an overwrite seems to be the wrong attack vector for the problem at hand.

Reply:
In comparison to the functionality of the org.eclipse.ui.editors.text.TextFileDocumentProvider we see that they force overwrite in the method org.eclipse.ui.editors.text.TextFileDocumentProvider.createFileFromDocument(IProgressMonitor, IFile, IDocument) . 

Also JDT forces the write if the file is synchronized.. 
org.eclipse.jdt.internal.ui.javaeditor.CompilationUnitDocumentProvider.commitWorkingCopy(IProgressMonitor, Object, CompilationUnitInfo, boolean)

I've adjusted my solution similar to JDT .. I hope this is fine now.